portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,18 +30,45 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet { 
 
+  ArrayList<String> comments = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> comments = new ArrayList<String>();
-    comments.add("Hello");
-    comments.add("What's up?");
-    comments.add("Good Morning");
-
-    String json =convertToJsonUsingGson(comments);
+    String json = convertToJsonUsingGson(comments);
```

Comment:
Isis, I think you are most of the way there. The way this code looks now, it looks like the comments variable you have is only persisting them in memory for the session, whereas in the doPost method below you are storing them permanently.

I think you can rework this method to read from the datastore Service, just like how you are putting them in the datastore service below.

```
DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
PreparedQuery results = datastore.prepare(query);
```

something along those lines. 

Feel free to reach out if this comment needs clarification :)

Reply:
+1